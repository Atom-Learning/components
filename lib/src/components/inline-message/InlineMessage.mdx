---
title: Inline Message
component: InlineMessage
description: Inline feedback message with an optional icon
category: Feedback
---

Inline messages consist of an inline feedback message in the form of simple text that inform the user of relevant information, revealed in context.

```tsx preview
<InlineMessage>This is an Inline Message</InlineMessage>
```

## Theme

These are the available `theme`s for this component: `success`, `warning`, `error`, `system` and `info`. The default is `error` (due to most frequent context).

```tsx preview
<>
  <InlineMessage theme="error">This is an error Inline Message</InlineMessage>
  <InlineMessage theme="warning">
    This is a warning Inline Message
  </InlineMessage>
  <InlineMessage theme="success">
    This is a success Inline Message
  </InlineMessage>
  <InlineMessage theme="info">This is an info Inline Message</InlineMessage>
  <InlineMessage theme="system">This is a system Inline Message</InlineMessage>
</>
```

## Size

These are the available `size`s for this component: `xs`, `sm` and `md`. The default is `sm`

```tsx preview
<>
  <InlineMessage size="xs">This is an xs Inline Message</InlineMessage>
  <InlineMessage size="sm">This is a sm Inline Message</InlineMessage>
  <InlineMessage size="md">This is a md Inline Message</InlineMessage>
</>
```

## Icon

Each theme has a specific icon it should export, so showing or hiding an icon is simply done using the `showIcon` prop. Default is `true`.

```tsx preview
<InlineMessage showIcon={false}>This is an Inline Message</InlineMessage>
```
