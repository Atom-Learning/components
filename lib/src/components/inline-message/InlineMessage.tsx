import * as React from 'react'

import { Flex } from '~/components/flex'
import { Icon } from '~/components/icon'
import { Text } from '~/components/text'
import { styled } from '~/stitches'

import {
  INLINE_MESSAGE_ICONS,
  INLINE_MESSAGE_THEMES
} from './InlineMessage.config'
import { InlineMessageTheme } from './InlineMessage.types'

const InlineMessageContainer = styled(Flex, {
  variants: {
    theme: INLINE_MESSAGE_THEMES
  }
})

type TInlineMessageProps = React.ComponentProps<
  typeof InlineMessageContainer
> & {
  showIcon?: boolean
  size?: 'xs' | 'sm' | 'md'
}

export const InlineMessage = ({
  css,
  showIcon = true,
  theme = 'error',
  size = 'sm',
  children,
  ...rest
}: TInlineMessageProps) => (
  <InlineMessageContainer theme={theme} css={css} {...rest}>
    {showIcon && (
      <Icon
        css={{ mr: '$2' }}
        size="sm"
        is={INLINE_MESSAGE_ICONS[theme as InlineMessageTheme]}
      />
    )}
    <Text as="span" size={size} css={{ transform: 'translateY($space$0)' }}>
      {children}
    </Text>
  </InlineMessageContainer>
)
