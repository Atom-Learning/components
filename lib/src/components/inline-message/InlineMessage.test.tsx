import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { InlineMessage } from './InlineMessage'

describe(`InlineMessage component`, () => {
  it('renders an inline message', async () => {
    const { container } = render(
      <InlineMessage>This is an Inline Message</InlineMessage>
    )

    await screen.getByText('This is an Inline Message')

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <InlineMessage>This is an Inline Message</InlineMessage>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('does not render icon when showIcon prop is false', async () => {
    const { container } = render(
      <InlineMessage showIcon={false}>This is an Inline Message</InlineMessage>
    )

    const icon = container.querySelector('svg')
    expect(icon).toBeFalsy()
  })
})
