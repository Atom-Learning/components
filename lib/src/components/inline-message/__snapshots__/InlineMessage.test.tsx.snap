// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineMessage component renders an inline message 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-fzYfTW {
    flex-shrink: 0;
    margin-right: var(--space-2);
  }

  .c-fIXJvv {
    color: var(--colors-tonal600);
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-fIXJvv > .c-fIXJvv:before,
  .c-fIXJvv > .c-fIXJvv:after {
    display: none;
  }
}

@media  {
  .c-PJLV-iymbh-theme-error {
    color: var(--colors-danger);
  }

  .c-dbrbZt-cyeZeh-size-sm {
    height: var(--sizes-1);
    width: var(--sizes-1);
    stroke-width: 1.5;
  }

  .c-fIXJvv-bndJoy-size-sm {
    font-size: var(--fontSizes-sm);
    line-height: 1.53;
  }

  .c-fIXJvv-bndJoy-size-sm::before {
    content: '';
    margin-bottom: -0.4056em;
    display: table;
  }

  .c-fIXJvv-bndJoy-size-sm::after {
    content: '';
    margin-top: -0.4056em;
    display: table;
  }
}

@media  {
  .c-fIXJvv-igszWir-css {
    color: inherit;
    transform: translateY(var(--space-0));
  }
}

<div>
  <div
    class="c-dhzjXW c-PJLV c-PJLV-iymbh-theme-error"
  >
    <svg
      aria-hidden="true"
      class="c-dbrbZt c-dbrbZt-cyeZeh-size-sm c-fzYfTW"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 8v5"
      />
      <line
        x1="12"
        x2="12"
        y1="16"
        y2="16"
      />
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
    <span
      class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-fIXJvv-igszWir-css"
    >
      This is an Inline Message
    </span>
  </div>
</div>
`;
